src/lib.rs

```diff
@@ -1368,6 +1380,157 @@ where
     }
 }
 
+#[derive(Clone, Copy, Serialize, Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord)]
+/// A group of `len_items` items, from shard `shard`, stored at position `offset`, using `len_bytes` bytes on-disk.
+/// Similar to ShardRecord, just that we don't store the key. Used for `UnsortedShardReader`
+struct KeylessShardRecord {
+    offset: usize,
+    len_bytes: usize,
+    len_items: usize,
+}
+
+/// Read from a collection of shardio files in the order in which items are written without
+/// considering the sort order.
+///
+/// Useful if you just want to iterate over all the items irrespective of the ordering.
+///
+#[allow(dead_code)]
+pub struct UnsortedShardReader<T, S = DefaultSort>
```

Comment:
Ran into an issue trying to give this a go to read our aligned shard files.  It looks like needing a sort is a bit of a complication in my case

```
error[E0277]: the trait bound `rust_htslib::bam::Record: Ord` is not satisfied
    --> pybam/src/lib.rs:24:13
     |
24   |     reader: UnsortedShardReader<Record>,
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Ord` is not implemented for `rust_htslib::bam::Record`
     | 
    ::: /mnt/bazelbuild/user/nigel.delaney/cargo/git/checkouts/rust-shardio-1f7cda12cec2799a/810508d/src/lib.rs:1400:8
     |
1400 |     S: SortKey<T>,
```
Thought to ask what best solution might be, do we really need to care about the ability to order unordered items?

Reply:
I see, yeah, with the current API you need to specify the sort order explicitly in the `UnsortedShardReader` as well. This can be relaxed and would need a bit of refactoring to pull out things in `ShardReaderSingle` that does not use the sort type.